src/Utilities/model.jl

```diff
@@ -137,16 +145,48 @@ end
 MOI.canget(model::AbstractModel, ::MOI.Name) = true
 MOI.get(model::AbstractModel, ::MOI.Name) = model.name
 
+"""
+    newname(model::MOI.ModelLike, IndexType::Type{<:MOI.Index}, idx::MOI.Index, name::String)
```

Comment:
`check_can_assign_name` maybe? It's a bit confusing why this method should throw an error. What about returning 1 of 3 possible statuses: `NameUnassigned`, `NameAssignedAndMatches`, and `NameMismatch`?

Reply:
If it does not throw an error, there won't be much code to share between the three places where this is used since we will have to branch on these three statuses. If we make typos in these branches we will introduce bugs and inconsistencies between the three places